RecoEgamma/PhotonIdentification/src/PhotonDNNEstimator.cc

```diff
@@ -0,0 +1,255 @@
+          type = 3;
+        else
+          type = 0;
+        features.push_back(std::make_tuple(varname, type, par1, par2));
+        auto match = std::find(
+            PhotonDNNEstimator::dnnAvaibleInputs.begin(), PhotonDNNEstimator::dnnAvaibleInputs.end(), varname);
+        if (match == std::end(PhotonDNNEstimator::dnnAvaibleInputs)) {
+          throw cms::Exception("MissingVariable")
+              << "Requested variable (" << varname << ") not available between Photon PFid DNN inputs";
+        }
+        ninputs += 1;
+      }
+    }
+    inputfile_scaler.close();
+    featuresMap_.push_back(features);
+    nInputs_.push_back(ninputs);
+  }
+}
+
+const std::array<std::string, PhotonDNNEstimator::nAvailableVars> PhotonDNNEstimator::dnnAvaibleInputs = {
+    {"hadTowOverEm",
+     "TrkSumPtHollow",
+     "EcalRecHit",
+     "SigmaIetaIeta",
+     "SigmaIEtaIEtaFull5x5",
+     "SigmaIEtaIPhiFull5x5",
+     "EcalPFClusterIso",
+     "HcalPFClusterIso",
+     "HasPixelSeed",
+     "R9Full5x5",
+     "hcalTower"}};
+
+std::map<std::string, float> PhotonDNNEstimator::getInputsVars(const reco::Photon& photon) const {
+  // Prepare a map with all the defined variables
+  std::map<std::string, float> variables;
+  variables["hadTowOverEm"] = photon.hadTowOverEmValid() ? photon.hadTowOverEm() : 0;
+  variables["TrkSumPtHollow"] = photon.trkSumPtHollowConeDR03();
+  variables["EcalRecHit"] = photon.ecalRecHitSumEtConeDR03();
+  variables["SigmaIetaIeta"] = photon.sigmaIetaIeta();
+  variables["SigmaIEtaIEtaFull5x5"] = photon.full5x5_sigmaIetaIeta();
+  variables["SigmaIEtaIPhiFull5x5"] = photon.full5x5_showerShapeVariables().sigmaIetaIphi;
+  variables["EcalPFClusterIso"] = photon.ecalPFClusterIso();
+  variables["HcalPFClusterIso"] = photon.hcalPFClusterIso();
+  variables["HasPixelSeed"] = (Int_t)photon.hasPixelSeed();
+  variables["R9Full5x5"] = photon.full5x5_r9();
+  variables["hcalTower"] = photon.hcalTowerSumEtConeDR03();
+  variables["R9Full5x5"] = photon.full5x5_r9();
+  // Define more variables here and use them directly in the model config!
+  return variables;
+}
+
+uint PhotonDNNEstimator::getModelIndex(const reco::Photon& photon) const {
+  /* 
+  Selection of the model to be applied on the Photon based on pt/eta cuts or whatever selection
+  */
+  uint modelIndex;
+  if (std::abs(photon.eta()) <= 1.466) {
+    modelIndex = 0;
+  } else {
+    modelIndex = 1;
+  }
+  return modelIndex;
+}
+
+std::pair<uint, std::vector<float>> PhotonDNNEstimator::getScaledInputs(const reco::Photon& photon) const {
+  uint modelIndex = getModelIndex(photon);
+  auto allInputs = getInputsVars(photon);
+  std::vector<float> inputs;
+  // Loop on the list of requested variables and scaling values for the specific modelIndex
+  // Different type of scaling are available: 0=noscaling, 1=standard scaler, 2=minmax
+  for (auto& [varName, type, par1, par2] : featuresMap_[modelIndex]) {
```

Comment:
I notice that the scaler is duplicated between the electron and photon estimator. Can this be put in a small class to reduce code duplication?